### PR TITLE
Allow parameter to be used with HotKey action

### PR DESCRIPTION
**Pull request by kphrx**
This change allows you to take advantage of the arguments passed to the callback.

```diff
 func applicationDidFinishLaunching(_ aNotification: Notification) {
     // ⌘ + Control + B
     guard let keyCombo: KeyCombo = KeyCombo(keyCode: 11, carbonModifiers: 4352) else { return }
     let hotKey: HotKey = HotKey(identifier: "CommandControlB",
                                 keyCombo: keyCombo,
                                 target: self,
-                                action: #selector(AppDelegate.handleHotKeyEvent))
+                                action: #selector(AppDelegate.handleHotKeyEvent(_:)))
     hotKey.register()
 }

-@objc func handleHotKeyEvent() {
+@objc func handleHotKeyEvent(_ hotKey: HotKey) {
-    print("CommandControlB") // CommandControlB
+    let identifier = hotKey.identifier
+    print(identifier) // CommandControlB
 }
```

### CODE REVIEW

**Comment by kphrx** [comment]
This function is possible by using the closure supported by this PR (#20).

```diff
 func applicationDidFinishLaunching(_ aNotification: Notification) {
     // ⌘ + Control + B
     guard let keyCombo: KeyCombo = KeyCombo(keyCode: 11, carbonModifiers: 4352) else { return }
     let hotKey: HotKey = HotKey(identifier: "CommandControlB",
                                 keyCombo: keyCombo,
-                                target: self,
-                                action: #selector(AppDelegate.handleHotKeyEvent(_:)))
+                                handler: handleHotKeyEvent)
     hotKey.register()
 }

-@objc func handleHotKeyEvent(_ hotKey: HotKey) {
+func handleHotKeyEvent(_ hotKey: HotKey) {
     let identifier = hotKey.identifier
     print(identifier) // CommandControlB
 }
```

File: Lib/Magnet/HotKey.swift [hunk 1]
```diff
@@ -9,7 +9,8 @@
 import Cocoa
 import Carbon
 
-public final class HotKey: Equatable {
+@objc(HotKey)
```
Comment (Econa77) [inline Lib/Magnet/HotKey.swift]: Is this `@objc` annotation necessary?

File: Lib/Magnet/HotKey.swift [hunk 2]
```diff
@@ -9,7 +9,8 @@
...
+public final class HotKey: NSObject {
```
Comment (Econa77) [inline Lib/Magnet/HotKey.swift:12]: Please change `Equatable` of `NSObject` must override `isEqual(_:)`.
Like this https://github.com/Clipy/Magnet/blob/33a13816b27e20e39e580af1e001099190caa7e3/Lib/Magnet/KeyCombo.swift#L84-L90

**Review by Econa77 — CHANGES_REQUESTED** [review]
@kPherox Sorry for late reply. I made some reviews so can you fix it?

File: Lib/Magnet/HotKey.swift [hunk 3]
```diff
@@ -41,6 +42,7 @@ public final class HotKey: Equatable {
 
     // MARK: - Initialize
     public init(identifier: String, keyCombo: KeyCombo, target: AnyObject, action: Selector, actionQueue: ActionQueue = .main) {
+        super.init()
```
Comment (Econa77) [inline Lib/Magnet/HotKey.swift]: Please call `super.init()` method at the end of initializer.

Comment (kphrx) [reply Lib/Magnet/HotKey.swift]: Fixed with a00a0ed

File: Lib/Magnet/HotKey.swift (hunk 2 above)
Comment (kphrx) [reply Lib/Magnet/HotKey.swift:12]: Changed to override isEqual with e5cf7f0

**Review by Econa77 — APPROVED** [review]
LGTM 👍 Thanks :)

File: Lib/Magnet/HotKey.swift (hunk 1 above)
Comment (kphrx) [reply Lib/Magnet/HotKey.swift]: No, this is unnecessary.

In order to be able to use `HotKey` in ObjC, after writing the `@objc` annotation, we replaced `Equatable` with `NSObject`. So I did not think it was unnecessary.

**Comment by Econa77** [comment]
Released [v2.3.1](https://github.com/Clipy/Magnet/releases/tag/v2.3.1) 🎉

**Pull request merged**